api/src/services/security-service.ts

```diff
@@ -330,53 +330,94 @@ export class SecurityService extends DBService {
   }
 
   /**
-   * Applies all given security rules to the given set of submission feature ids.
-   * This process is additive unless the override flag is set to `true`.
-   * If the override is set to `true` all security rules for the given set of features will be removed.
-   * Then the new security rules will be applied as normal.
+   * Patches security rules that are applied or removed to the given set of submission features. If a
+   * particular rule happens to belong to both `applyRuleIds` and `removeRuleIds`, it will always be
+   * added.
    *
-   * @param {number[]} features
-   * @param {number[]} rules
-   * @param {boolean}
-   * @return {*}  {Promise<SubmissionFeatureSecurityRecord[]>}
+   * @param {number[]} submissionFeatureIds IDs of the submission features whose security will be updated.
+   * @param {number[]} applyRuleIds IDs of the rules which will be applied after the patch operation
+   * @param {number[]} removeRuleIds IDs of the rules which will be removed after the patch operation
+   * @return {*}  {Promise<void>}
    * @memberof SecurityService
    */
-  async applySecurityRulesToSubmissionFeatures(
-    features: number[],
-    rules: number[],
-    override = false
-  ): Promise<SubmissionFeatureSecurityRecord[]> {
-    if (override) {
-      // we want to override any security rules present and can achieve this by remove everything first
-      await this.securityRepository.removeSecurityRulesFromSubmissionFeatures(features);
+  async patchSecurityRulesOnSubmissionFeatures(
+    submissionFeatureIds: number[],
+    applyRuleIds: number[],
+    removeRuleIds: number[]
+  ): Promise<void> {
+    defaultLog.debug({ label: 'patchSecurityRulesOnSubmissionFeatures', applyRuleIds, removeRuleIds });
+
+    if (!submissionFeatureIds.length) {
```

Comment:
Is this check required when our api doc declares that min 1 item is required by the PATCH request.

Reply:
To add to my previous comment, I think of the API schema as moreso a way to catch malformed queries made by the user, whereas this check is added to simply forgo making an empty request on the database.